src/test/javascript/spec/component/admin/health.component.spec.ts

```diff
@@ -6,61 +6,86 @@ import { ArtemisTestModule } from '../../test.module';
-        it('should get badge class', () => {
-            const upBadgeClass = comp.getBadgeClass('UP');
-            const downBadgeClass = comp.getBadgeClass('DOWN');
-            expect(upBadgeClass).toEqual('bg-success');
-            expect(downBadgeClass).toEqual('bg-danger');
-        });
+    it('should get badge class', () => {
+        const upBadgeClass = comp.getBadgeClass('UP');
+        const downBadgeClass = comp.getBadgeClass('DOWN');
+        expect(upBadgeClass).toEqual('bg-success');
+        expect(downBadgeClass).toEqual('bg-danger');
     });
 
-    describe('refresh', () => {
-        it('should call refresh on init', () => {
-            // GIVEN
-            const health: Health = { status: 'UP', components: { mail: { status: 'UP', details: { mailDetail: 'mail' } } } };
-            jest.spyOn(service, 'checkHealth').mockReturnValue(of(health));
+    it('should call refresh on init', () => {
+        // GIVEN
+        const health: Health = { status: 'UP', components: { mail: { status: 'UP', details: { mailDetail: 'mail' } } } };
+        jest.spyOn(healthService, 'checkHealth').mockReturnValue(of(health));
 
-            // WHEN
-            comp.ngOnInit();
+        // WHEN
+        comp.ngOnInit();
 
-            // THEN
-            expect(service.checkHealth).toHaveBeenCalled();
-            expect(comp.health).toEqual(health);
-        });
+        // THEN
+        expect(healthService.checkHealth).toHaveBeenCalledTimes(1);
+        expect(comp.health).toBe(health);
+    });
+
+    it('should handle a 503 on refreshing health data', () => {
+        // GIVEN
+        const health: Health = { status: 'DOWN', components: { mail: { status: 'DOWN' } } };
+        jest.spyOn(healthService, 'checkHealth').mockReturnValue(throwError(() => new HttpErrorResponse({ status: 503, error: health })));
+
+        // WHEN
+        comp.refresh();
+
+        // THEN
+        expect(healthService.checkHealth).toHaveBeenCalledTimes(1);
+        expect(comp.health).toBe(health);
+    });
+
+    it('should open a modal with health if eye icon is clicked', () => {
+        const health: Health = { status: 'UP', components: { mail: { status: 'UP', details: { mailDetail: 'mail' } } } };
+        jest.spyOn(healthService, 'checkHealth').mockReturnValue(of(health));
+
+        const mockModalRef = { componentInstance: {} } as NgbModalRef;
+        const modalServiceSpy = jest.spyOn(modalService, 'open').mockReturnValue(mockModalRef);
+
+        fixture.detectChanges();
 
-        it('should handle a 503 on refreshing health data', () => {
-            // GIVEN
-            const health: Health = { status: 'DOWN', components: { mail: { status: 'DOWN' } } };
-            jest.spyOn(service, 'checkHealth').mockReturnValue(throwError(() => new HttpErrorResponse({ status: 503, error: health })));
+        const linkToClick = fixture.debugElement.query(By.css('a.hand'));
+        expect(linkToClick).not.toBe(null);
```

Comment:
Is it possible to test for an expected property/field value here?

Reply:
Not that I'm aware of. This is just a test that the element queried through CSS exists, as suggested in the table at the bottom of https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/